hedera-node/hedera-app/src/main/java/com/hedera/node/app/hints/HintsLibrary.java

```diff
@@ -0,0 +1,108 @@
+import java.util.Map;
+
+/**
+ * The cryptographic operations required by the {@link HintsService}.
+ */
+public interface HintsLibrary {
+    /**
+     * Generates a new BLS key pair.
+     * @return the key pair
+     */
+    BlsKeyPair newBlsKeyPair();
+
+    /**
+     * Signs the given message with the given private key.
+     * @param message the message
+     * @param key the private key
+     * @return the signature
+     */
+    BlsSignature signPartial(@NonNull Bytes message, @NonNull BlsPrivateKey key);
+
+    /**
+     * Verifies the given signature for the given message and public key.
+     * @param message the message
+     * @param signature the signature
+     * @param publicKey the public key
+     * @return true if the signature is valid; false otherwise
+     */
+    boolean verifyPartial(@NonNull Bytes message, @NonNull BlsSignature signature, @NonNull BlsPublicKey publicKey);
+
+    /**
+     * Aggregates the signatures for the given party ids with the given aggregation key.
+     * @param aggregationKey the aggregation key
+     * @return the aggregated signature
+     */
+    Bytes aggregateSignatures(@NonNull Bytes aggregationKey, @NonNull Map<Long, BlsSignature> signatures);
+
+    /**
+     * Extracts the public key for the given party id from the given aggregation key.
+     * @param aggregationKey the aggregation key
+     * @param partyId the party id
+     * @return the public key
+     */
+    BlsPublicKey extractPublicKey(@NonNull Bytes aggregationKey, long partyId);
+
+    /**
+     * Extracts the weight of the given party id from the given aggregation key.
+     * @param aggregationKey the aggregation key
+     * @param partyId the party id
+     * @return the weight
+     */
+    long extractWeight(@NonNull Bytes aggregationKey, long partyId);
+
+    /**
+     * Extracts the total weight of all parties from the given aggregation key.
+     * @param aggregationKey the aggregation key
+     * @return the weight
+     */
+    long extractTotalWeight(@NonNull Bytes aggregationKey);
+
+    /**
+     * Computes the hints for the given public key and number of parties.
+     * @param privateKey the private key
+     * @param n the number of parties
+     * @return the hints
+     */
+    Bytes computeHints(@NonNull BlsPrivateKey privateKey, int n);
+
+    /**
+     * Validates the hinTS public key for the given number of parties.
+     *
+     * @param n the number of parties
+     * @return true if the hints are valid; false otherwise
+     */
+    boolean validate(@NonNull Bytes hintsKey, int n);
+
+    /**
+     * Aggregates the given validated hint keys and party weights into preprocessed keys.
+     * @param hintKeys the valid hint keys by party id
+     * @param weights the weights by party id
+     * @param n the number of parties
+     * @return the aggregated keys
+     */
+    Bytes preprocess(@NonNull Map<Long, Bytes> hintKeys, @NonNull Map<Long, Long> weights, int n);
```

Comment:
This will be changed to `PreprocessedKeys` after protobufs are merged. Just added comment in case others are reviewing. 🙂